arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -140,32 +141,36 @@ RocksDBEngine::RocksDBEngine(application_features::ApplicationServer* server)
 
 RocksDBEngine::~RocksDBEngine() {
   shutdownRocksDBInstance(); 
+
+  delete _db;
 }
 
 /// shuts down the RocksDB instance. this is called from unprepare
 /// and the dtor
 void RocksDBEngine::shutdownRocksDBInstance() noexcept {
-  if (_db) {
-    // turn off RocksDBThrottle, and release our pointers to it
-    if (nullptr != _listener.get()) {
-      _listener->StopThread();
-    } // if
-    
-    for (rocksdb::ColumnFamilyHandle* h : RocksDBColumnFamily::_allHandles) {
-      _db->DestroyColumnFamilyHandle(h);
-    }
-    
-    // now prune all obsolete WAL files
-    try {
-      determinePrunableWalFiles(0);
-      pruneWalFiles();
-    } catch (...) {
-      // this is allowed to go wrong on shutdown
-      // we must not throw an exception from here
-    }
+  if (_hasDoneShutdown || !_db) {
+    return;
```

Comment:
Your choice, but I have never enjoyed code with multiple return points.  This could have easily been "if (_db && !_hasDoneShutdown)".  That would leave the code flow exiting at bottom of function.  Easier to maintain in the long run.